geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroDataFile.scala

```diff
@@ -0,0 +1,37 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.features.avro
+
+import org.apache.avro.file.{DataFileStream, DataFileWriter}
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+object AvroDataFile {
```

Comment:
I think there's some merit :) Essentially the avro files are:
* a binary format for long term storage of simple features or transit between systems. Since its binary it's more efficient than XML, json, etc. 
* schemad with metadta - Since it is schema'd and has metadata it carries the SFT with it. 
* Standalone-ish they can be ingested to a new catalog without knowing the SFT (like a shapefile)